Samples/src/OracleEmulation/Operations.qs

```diff
@@ -0,0 +1,162 @@
+            oracle([x], y);
+            MeasureAndDisplay("The answer is ", y);
+
+            // Apply an oracle that was declared as `intrinsic` above and
+            // implemented in the C# driver.
+            HalfAnswer([x], y);
+            MeasureAndDisplay("Half the answer is ", y);
+
+            // Apply an oracle defined in terms of a Q# permutation function.
+            EmulateOracle(DoubleAnswerFunc, [x], y);
+            MeasureAndDisplay("Twice the answer is ", y);
+
+            // Apply an oracle to a superposition in y.
+            for(i in 1..5) {
+                H(y[7]);
+                // Before the oracle is queried, the state of the y register is
+                //      $\ket{y} = \ket{0} + \ket{128}$.
+                // The oracle will map this to
+                //      $\ket{y'} = \ket{42 \oplus 0} + \ket{42 \oplus 128} = \ket{42} + \ket{170}$.
+                oracle([x], y);
+                MeasureAndDisplay("The answer might be ", y);
+            }
+
+            Reset(x);
+        }
+    }
+
+
+    //////////////////////////////////////////////////////////////////////////
+    // Emulated arithmetic operations ////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////
+
+    // Prepare two `LittleEndian` registers in a computational basis state.
+    operation PrepareSummands(qubits: Qubit[], numbers: Int[]) : (LittleEndian, LittleEndian) {
+        let n = Length(qubits);
+        let x = LittleEndian(qubits[0..n/2 - 1]);
+        let y = LittleEndian(qubits[n/2..n-1]);
+        InPlaceXorLE(numbers[0], x);
+        InPlaceXorLE(numbers[1], y);
+        return (x, y);
+    }
+
+    // Measure and check that M(x) + y_init == M(y).
+    operation MeasureAndCheckAddResult(x: LittleEndian, y: LittleEndian, y_init: Int): (Int, Int) {
+        let mx = MeasureInteger(x);
+        let my = MeasureInteger(y);
+        Message($"Computed {mx} + {y_init} = {my} mod {2^8}");
+        AssertBoolEqual((mx + y_init) % 2^8 == my, true, "sum is wrong");
+        return (mx, my);
+    }
+
+    // Modular addition of 8-bit integers.
+    function ModAdd8(x: Int, y: Int) : Int {
+        return (x + y) % (1 <<< 8);
+    }
+
+    // Here we demonstrate how to define and use emulated arithmetic operations.
+    // Our example is modular addition of 8-bit integers as implemented by the
+    // one-line function above. This defines already a permutation function on
+    // the computational basis states of two 8-qubit registers:
+    //      $\ket{m}\ket{n} \rightarrow \ket{m}\ket{m + n \mod 8}$.
+    // We can hence directly turn this function into an emulated oracle.
+    operation RunAddOracle() : Unit {
+        Message("Running emulated addition...");
+
+        // Turn the permutation function into an oracle operation acting on two
+        // quantum registers.
+        let adder = EmulateOracle(ModAdd8, _, _);
```

Comment:
Similar comment with respect to operationalism here.

Reply:
Should be resolved as per discussion above.